### PR TITLE
Column data model 9

### PR DESCRIPTION
**Pull request by jacobp100**

### CODE REVIEW

File: packages/ag-charts-community/src/chart/data/dataModel.test.ts [hunk 1]
```diff
@@ -643,35 +599,29 @@ describe('DataModel', () => {
...
-                expect(result?.data.length).toEqual(2);
-                expect(result?.data[0].keys).toEqual(['Q1']);
-                expect(result?.data[1].keys).toEqual(['Q2']);
+                expect(result.type).toEqual('grouped');
+                expect(result.groups).toHaveLength(2);
+                expect(result.groups[0].keys).toEqual(['Q1']);
+                expect(result.groups[1].keys).toEqual(['Q2']);
             });
 
             it('should extract the configured values', () => {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/data/dataModel.test.ts:610]: This test case seems to have dropped some assertions?

File: packages/ag-charts-community/src/chart/data/dataModel.test.ts [hunk 2]
```diff
@@ -742,35 +688,29 @@ describe('DataModel', () => {
...
-                expect(result?.data.length).toEqual(2);
-                expect(result?.data[0].keys).toEqual(['Q1']);
-                expect(result?.data[1].keys).toEqual(['Q2']);
+                expect(result.type).toEqual('grouped');
+                expect(result.groups).toHaveLength(2);
+                expect(result.groups[0].keys).toEqual(['Q1']);
+                expect(result.groups[1].keys).toEqual(['Q2']);
             });
 
             it('should extract the configured accumulated values', () => {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/data/dataModel.test.ts:699]: Ditto

File: packages/ag-charts-community/src/chart/data/dataModel.test.ts [hunk 3]
```diff
@@ -1171,7 +1089,7 @@ describe('DataModel', () => {
 `);
         });
 
-        describe('property tests', () => {
+        describe.skip('property tests', () => {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/data/dataModel.test.ts:1092]: Nit: `.skip()?`

File: packages/ag-charts-community/src/chart/data/dataModel.test.ts (hunk 1 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/data/dataModel.test.ts:610]: I've not been super consistent with the testing here. Some of the tests take these indices and then read from the columns to compare the data. This just assumes the columns are correct (they're tested elsewhere), so here it's just validating you'll read the right location in the column. It's sort of the same thing, but I can update it if you feel strongly?

File: packages/ag-charts-community/src/chart/data/dataModel.test.ts (hunk 2 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/data/dataModel.test.ts:699]: Same - https://github.com/ag-grid/ag-charts/pull/2914/files/c3031a99beb3e71d9c0c28a1f7b73ea47d1932c6#r1829530633

File: packages/ag-charts-community/src/chart/data/dataModel.test.ts (hunk 3 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/data/dataModel.test.ts:1092]: Fixed in 10

File: packages/ag-charts-community/src/chart/data/dataModel.test.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/data/dataModel.test.ts:610]: Ok, if the values are covered elsewhere then SGTM

**Review by alantreadway — APPROVED** [review]

**Pull request closed without merging**